modules/programs/hexchat.nix

```diff
@@ -0,0 +1,323 @@
+      (transformField "i" channel.nickname2)
+      (transformField "R" channel.realName)
+      (transformField "U" channel.userName)
+      (transformField "P" channel.password)
+      (listChar "S" channel.servers)
+      (listChar "J" channel.autojoin)
+      (listChar "C" channel.commands)
+    ]));
+in {
+  meta.maintainers = with maintainers; [ superherointj thiagokokada ];
+
+  options.programs.hexchat = with types; {
+    enable = mkEnableOption "HexChat - Graphical IRC client";
+
+    channels = mkOption {
+      default = null;
+      type = nullOr (attrsOf modChannelOption);
+      description = "Configures '~/.config/hexchat/servlist.conf'";
+      example = literalExample ''
+        {
+          freenode = {
+            autojoin = [
+              "#home-manager"
+              "##linux"
+              "#nixos"
+            ];
+            charset = "UTF-8 (Unicode)";
+            commands = [
+              "ECHO Buzz Lightyear sent you a message: 'To Infinity... and Beyond!'"
+            ];
+            loginMethod = sasl;
+            nickname = "my_nickname";
+            nickname2 = "my_secondchoice";
+            options = {
+              acceptInvalidSSLCertificates = false;
+              autoconnect = true;
+              bypassProxy = true;
+              connectToSelectedServerOnly = true;
+              useGlobalUserInformation = false;
+              forceSSL = false;
+            };
+            password = "my_password";
+            realName = "my_realname";
+            servers = [
+              "chat.freenode.net"
+              "irc.freenode.net"
+            ];
+            userName = "my_username";
+          };
+        }'';
+    };
+
+    settings = mkOption {
+      default = null;
+      description = ''
+        Configuration for "~/.config/hexchat/hexchat.conf", see
+        <link xlink:href="https://hexchat.readthedocs.io/en/latest/settings.html#list-of-settings"/>
+        for supported values.
+      '';
+      example = literalExample ''
+        {
+          irc_nick1 = "mynick";
+          irc_username = "bob";
+          irc_realname = "Bart Simpson";
+          text_font = "Monospace 14";
+        };
+      '';
+      type = nullOr (attrsOf str);
+    };
+
+    overwriteConfigFiles = mkOption {
+      type = nullOr bool;
+      description = ''
+        Enables overwritting HexChat configuration files (hexchat.conf, servlist.conf).
+        Any existing HexChat configuration will be lost.
+        Certify to back-up any previous configuration before enabling this.
+
+        Enabling this setting is recommended, because everytime HexChat application is
+        closed it overwrites Nix/Home-Manager provided configuration files, causing:
+        1. Nix/HM provided configuration to be out of sync with actual active HexChat configuration.
+        2. Blocking Nix/HM updates until configuration files are manually removed.
+      '';
+      default = false;
+    };
```

Comment:
What is happening here is that hexchat is overwriting the link file for hexchat configuration. This is far from a good solution though, since if you change something in the interface the configuration is updated, only to be lost in the next `home-manager switch`.

Is there some other solution? Ideally something that made hexchat symlinks "immutable" (maybe removing the write permission from them).

Reply:
On using home-manager the single source of truth becomes home-manager configuration not HexChat configuration GUI.

Reply:
Tried to set `hexchat.conf` to read-only (400) and even them it overwrites, so it seems that it is deleting and writing the file again.

Reply:
@thiagokokada is there an upstream bug/feature request to track?

Reply:
Just opened an issue: https://github.com/hexchat/hexchat/issues/2541

Reply:
As it is not clear there is a better upstream solution for now (as stated in upstream issue). This PR could be merged and later when a better solution is available a new PR can be done. The way it is now works. I have been using it.

Reply:
Some people at the issue above answered that we could use `chattr +i` to make the file immutable (I think this is the same as `/nix/store`). This seems like a good idea.

Is there any reason we don't set immutable flag in files managed by Home-Manager :thinking: ?

CC @berbiche @rycee .

Reply:
BTW, I think the current solution is ok (not perfect). It is definitely usable though. Unless we have some plans to set immutable files with Home-Manager itself I think this PR can be reviewed.

CC @berbiche.

Reply:
Hi, sorry for the late reply, I've been pretty busy recently.

The chattr option could be done with a custom activation script that runs after the files are linked.
AFAIK chattr options are filesystem-dependent so the script shouldn't rely on it.